ssh/tailssh/tailssh.go

```diff
@@ -122,19 +125,37 @@ func (srv *server) newSSHServer() (*ssh.Server, error) {
 		},
 		Version:                     "SSH-2.0-Tailscale",
 		LocalPortForwardingCallback: srv.mayForwardLocalPortTo,
-		NoClientAuthCallback: func(m gossh.ConnMetadata) (*gossh.Permissions, error) {
-			if srv.requiresPubKey(m.User(), toIPPort(m.LocalAddr()), toIPPort(m.RemoteAddr())) {
-				return nil, errors.New("public key required") // any non-nil error will do
+		NoClientAuthCallback: func(conn gossh.ConnMetadata) (*gossh.Permissions, error) {
+			var err error
+			ci, err = srv.connInfo(conn.User(), toIPPort(conn.LocalAddr()), toIPPort(conn.RemoteAddr()))
```

Comment:
This is just one `:=` refactoring away to silently break the code below where `ci == nil` on line 144 will then be true.

Hmm. Wish we had better tests here.

Reply:
Created a new struct for this. PTAL